src/QsCompiler/LanguageServer/LanguageServer.cs

```diff
@@ -454,5 +483,42 @@ bool FileIsWithinProjectDir(Uri projFile)
             foreach (var fileEvent in changes.Where(IsDll))
             { _ = this.EditorState.AssemblyDidChangeOnDiskAsync(fileEvent.Uri); }
         }
+
+        [JsonRpcMethod(Methods.TextDocumentCompletionName)]
+        public async Task<CompletionList> OnTextDocumentCompletionAsync(JToken arg)
+        {
+            // Wait for the file manager to finish processing any changes that happened right before this completion
+            // request.
+            await Task.Delay(50);
+            try
+            {
+                return QsCompilerError.RaiseOnFailure(
+                    () => EditorState.Completions(Utils.TryJTokenAs<TextDocumentPositionParams>(arg)),
+                    "Completions threw an exception");
+            }
+            catch
+            {
+                return null;
```

Comment:
Did you try if everything looks nice even in the case where null is returned? In particular in VS, I've had it before that returning null in some cases caused trouble and/or things looked a bit awkward. Could you please double check this? Same for ResolveCompletion below.

Reply:
It seems to work fine in the sense that if null is returned, the completion window doesn't show/no documentation is resolved.

Reply:
Perfect!